src/irmin-http/swagger.yml

```diff
@@ -17,90 +17,193 @@ paths:
+          description: "branch already exists, no-op"
         201:
-          description: "" # TODO
+          description: "success"
   /branch/{branchName}:
+    parameters:
+    - in: "path"
+      name: "branchName"
+      description: "name of branch to operate on"
+      required: true
+      type: "string"
     get:
       summary: "get a branch by name"
+      consumes:
+      - "application/json"
+      produces:
+      - "application/json"
       responses:
         200:
-          description: "" #TODO
+          description: "success"
+          schema:
+            $ref: "#/definitions/Branch"
         404:
-          description: "" #TODO
+          description: "branch not found"
     delete:
       summary: "delete a branch by its ID"
+      consumes:
+      - "application/json"
+      produces:
+      - "application/json"
       responses:
         200:
-          description: "" #TODO
+          description: "success"
         404:
-          description: "" #TODO
+          description: "branch not found"
   /commits:
     get:
-      summary: "lists all commits"
+      summary: "lists all commits, newest to oldest (like `git log`)"
+      consumes:
+      - "application/json"
+      produces:
+      - "application/json"
       responses:
         200:
-          description: "" #TODO
-        404:
-          description: "" #TODO
+          description: "success"
+          schema:
+            type: array
+            items:
+              $ref: "#/definitions/Commit"
   /commits/{commitHash}:
+    parameters:
+    - in: "path"
+      name: "commitHash"
+      description: "hash of commit to operate on"
+      required: true
+      type: "string"
     get:
       summary: "get a commit by its hash"
+      consumes:
+      - "application/json"
+      produces:
+      - "application/json"
       responses:
         200:
-          description: "" #TODO
+          description: "success"
+          schema:
+            $ref: "#/definitions/Commit"
         404:
-          description: "" #TODO
-  /paths:
+          description: "commit not found"
+  /keys:
     get:
       summary: "gets all keys and their associated values (like `irmin tree`)"
+      consumes:
+      - "application/json"
+      produces:
+      - "application/json"
       responses:
         200:
-          description: "" #TODO
-        404:
-          description: "" #TODO
+          description: "success"
+          schema:
+            type: array
+            items:
+              $ref: "#/definitions/KV"
     post:
       summary: "add a new key/value pair to the store"
+      consumes:
+      - "application/json"
+      produces:
+      - "application/json"
+      parameters:
+      - in: body
+        name: body
+        description: "key/value pair to add"
+        required: true
+        schema:
+          $ref: "#/definitions/KV"
       responses:
         200:
-          description: "" #TODO
+          description: "key already exists, no-op"
```

Comment:
204 HTTP code can be a solution (like: "OK but I will response nothing because we have nothing to do").

I just appear like a magic rabbit, you can miss my comment.